test/lightning_web/live/audit_live_test.exs

```diff
@@ -60,4 +61,296 @@ defmodule LightningWeb.AuditLiveTest do
       assert html =~ LiveHelpers.display_short_uuid(user_to_be_deleted.id)
     end
   end
+
+  describe ".diff/1" do
+    test "correctly lists changes with both before and after (string keys)" do
+      assigns = %{
+        metadata: %{
+          before: %{
+            "foo" => "foo_before",
+            "bar" => "bar_before"
+          },
+          after: %{
+            "foo" => "foo_after",
+            "bar" => "bar_after"
+          }
+        }
+      }
+
+      html = render_component(&AuditLive.Index.diff/1, assigns)
+
+      assert html =~ ~r"<li>foo.+foo_before.+foo_after</li>"s
+      assert html =~ ~r"<li>bar.+bar_before.+bar_after</li>"s
+    end
+
+    test "correctly lists changes with both before and after (atom keys)" do
```

Comment:
Having a string variant and an atom variant of each test would usually be overkill. However, I needed them to support the initial changes (ensuring consistent behaviour for strings and atoms) and, as they are lightweight and already 'paid for' :) we may as well keep them.